AsyncDisplayKit/ASMapNode.mm

```diff
@@ -279,7 +279,7 @@ - (void)setAnnotations:(NSArray *)annotations
 #pragma mark - Layout
 - (void)setSnapshotSizeWithReloadIfNeeded:(CGSize)snapshotSize
 {
-  if (!CGSizeEqualToSize(self.options.size, snapshotSize)) {
+  if (snapshotSize.height > 0 && snapshotSize.width > 0 && !CGSizeEqualToSize(self.options.size, snapshotSize)) {
```

Comment:
The exception is thrown immediately when you set it on the options object though. =/